### DISCUSSION TITLE
Single and Multi Selection

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/iTwinUI-react#723] [Q&A]
According to the user requirements in my application, we need Single Selection and Multi Selection. 
Here is the a snapshot of the table.
![image](https://user-images.githubusercontent.com/89520756/176366666-507fc49e-8960-40a1-ad17-098dc0b9d468.png)

Now there are two types of rows in the table, Parent and Sub rows. The Parent rows follow only Single Selection, However, Child rows or Subrows (as you can see in the snapshot) will have multi selection. So that multiple sub models can be selected. 
I am having trouble implementing this, I tried doing multi selection overall, but the checkboxes remained checked. As you can see in this SS.
![image](https://user-images.githubusercontent.com/89520756/176367435-c1c959d1-16a6-4977-a73b-eb446552b07e.png)

Kindly guide me regarding the best approach to implement this.

**Comment by veekeys** [comment]
Hello @hanan-commits-work,

We do not have this specific example, but you want to have fully custom selection behavior.
In that case, we usually recommend users to handle table state manually.
You would need to add reducer and handle events yourself.
We have small story as an example of that:
https://itwin.github.io/iTwinUI-react/?path=/story/core-table--controlled-state
![image](https://user-images.githubusercontent.com/81580355/176883093-b8f913bc-4e55-4a1d-9e56-42561e2b2fe7.png)


You should be able to catch different selection events, check the data and return the table state you want to have (manual handling of single selection and multiple selection).
